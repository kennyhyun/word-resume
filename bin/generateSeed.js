#! /usr/bin/env node

const fs = require('fs');
const { promisify } = require('util');
const yaml = require('js-yaml');

const generateSeed = require('../lib/seeder.js');

(async () => {
  const [,,outputFile] = process.argv;
  const writeFile = promisify(fs.writeFile);
  const d = generateSeed(`${new Date()}`);
  await writeFile(outputFile || 'sample.yml', yaml.dump(d));
})();
