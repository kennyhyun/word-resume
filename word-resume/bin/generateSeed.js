#! /usr/bin/env node

const fs = require('fs');
const { promisify } = require('util');
const yaml = require('js-yaml');

const generateSeed = require('../lib/seeder.js');

const [,,outputFile] = process.argv;
const writeFile = promisify(fs.writeFile);
generateSeed(`${new Date()}`, new Date(), outputFile || 'sample.yml');
