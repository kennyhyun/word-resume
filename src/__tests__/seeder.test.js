import generateSeed from '../seeder';

describe('generateSeed', () => {
  it('should seed sample yml', () => {
    const now = new Date('20i23-02-01');
    const source = generateSeed('jest test context', now);
    expect(source).toMatchSnapshot();
  });
});


