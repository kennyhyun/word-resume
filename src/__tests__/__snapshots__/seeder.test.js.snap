// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSeed should seed sample yml 1`] = `
Object {
  "data": Object {
    "educationHistory": Array [
      Object {
        "acquisition": "Invalid date",
        "duration": "Invalid date",
        "institute": "Idrudco University",
        "location": "Ciesouku, MS",
        "major": "Microbiologist Engineering",
        "title": "Bachelor of Engineering",
      },
    ],
    "language": Array [
      "Llamasese",
      "Lizardsese",
      "Caterpillarese",
    ],
    "profile": Object {
      "address": "1804 Usle Court",
      "carrierSummary": Object {
        "default": "Pebunmo uwpub ves uleoluove juc wowu er ofjo pi mazavehaf soni fezirwa foj hu. Efgi abisawog esoguojo zup hitmawe dizpaf ug kup porote litu limata kucu.",
        "frontend": "Vebemluw otismar dovwiglu dor gol bavu ze moja titokla efuibe loztenar ozwodka vizvo pij atniwez va. Pelmu at zimiv jousibig ik bu du lu hip ga ocvu hiir eh.",
      },
      "email": "tukuvfo@iwiefeuve.uy",
      "name": "Flora Chapman",
      "phone": "(868) 783-7248",
      "technicalSkills": Object {
        "generalOfficeSkills": Array [
          "petru",
          "koc",
          "ufvo",
          "olu",
          "nuhokun",
          "rumerfe",
          "fijud",
        ],
        "programmingLanguages": Array [
          "kulang",
          "kalang",
          "coplang",
        ],
      },
      "title": "Microbiologist",
    },
    "workHistory": Array [
      Object {
        "achievements": Array [],
        "company": "IDT Corporation",
        "duration": "Invalid date ~ Invalid date",
        "field": "Opi mikcavhad ojiegfa.",
        "location": "Odivedmi",
        "skills": Array [
          "ti",
          "avili",
          "idakeru",
          "nictuz",
          "kiwam",
          "hemloeh",
        ],
        "title": "Microbiologist",
        "type": "contract",
      },
      Object {
        "achievements": Array [
          "Mejce javnesih wo sapgud leocnor bojjig man doviki hirmezi nekedipo uctu foirla inpigav.",
          "Gi sivbug niwo ufutcu jinatpam juufa adikukkan tu guslaf tov muha amofogsog gudjit.",
          "Wuhepo vepu ca riwhewfel ped ulila uneira pooba atmu uteotu mem ceeji lulci.",
        ],
        "company": "State Street Corp.",
        "duration": "Invalid date ~ Invalid date",
        "field": "Ike bemozve zud.",
        "location": "Fohlefen",
        "skills": Array [
          "ruka",
          "demhevi",
          "pak",
        ],
        "title": "Microbiologist",
        "type": "contract",
      },
      Object {
        "achievements": Array [
          "Gilna de nicozig nibe ilwapeh fijtihom uwoufi mo ketmipes ti wopsu puvagum wa.",
          "Tekow luzose pamupuk najo ahrosmon pot efomuvzo vumkuvola ildut pegwaj boreblu sem kondedi.",
        ],
        "company": "McCormick & Company Incorporated",
        "duration": "Invalid date ~ Invalid date",
        "field": "Odwa otrima fejur.",
        "location": "Kizewe",
        "skills": Array [
          "fetinba",
          "hu",
          "amazo",
          "ezi",
          "mavvago",
          "dev",
          "to",
          "ilesi",
          "fitaben",
        ],
        "title": "Microbiologist",
        "type": "contract",
      },
      Object {
        "achievements": Array [
          "Recus acuepeer sit ojgeapo pajwe udar kumpaggo ise huhpahom ro dawbeb keotari vactase.",
          "Dazo hiziweweb tu mofilu lu epijof et dafik fuzdo nedooso esofmej sa jo.",
          "Cutjo iheozcem edukaco voij ucozo robimu tuvebi uc faccesit ogolab tiru du agegedaj.",
        ],
        "company": "Coca-Cola Co.",
        "duration": "Invalid date ~ Invalid date",
        "field": "Lu funludhu jo.",
        "location": "Cunahij",
        "skills": Array [
          "cezumurim",
          "vucnezuh",
          "buvhour",
          "tugsasa",
          "sejtoho",
          "ma",
          "si",
          "leacaho",
        ],
        "title": "Microbiologist",
        "type": "permanent",
      },
      Object {
        "achievements": Array [
          "Ohisoc covhaabu bu ri lezju anakuebe tuvne ugudi avzi oro fuwidken piabani co.",
          "Teresta co tacub rip mupgif led pawrimag nesu ho hi zew tawod ima.",
        ],
        "company": "Lithia Motors Inc.",
        "duration": "Invalid date ~ Invalid date",
        "field": "Ucuzel owmis sanvoh.",
        "location": "Efmoat",
        "skills": Array [
          "bocbupjoj",
          "keluga",
          "giw",
          "wesun",
          "mel",
          "ake",
        ],
        "title": "Microbiologist",
        "type": "contract",
      },
    ],
  },
  "header": Object {
    "author": "Flora Chapman",
    "description": "sample seed data generated by seed",
    "paragraphs": Array [
      Object {
        "source": "profile",
        "title": "$name",
      },
      Object {
        "digestRest": true,
        "first": 6,
        "source": "workHistory",
        "title": "Work History",
      },
      Object {
        "first": 4,
        "source": "educationHistory",
        "title": "Education",
      },
      Object {
        "source": "language",
        "title": "language",
      },
    ],
    "title": "Resume",
  },
}
`;
